integration/generic-tests/dhclient_test.go

```diff
@@ -85,25 +72,52 @@ func TestDhclient(t *testing.T) {
 // TestPxeboot runs a server and client to test pxebooting a node.
 // TODO: FIX THIS TEST!
 // Change the t.Logf below back to t.Errorf
-func TestPxeboot(t *testing.T) {
+func RunTestPxeboot(t *testing.T, initramfs string) {
 	// TODO: support arm
 	if vmtest.TestArch() != "amd64" {
 		t.Skipf("test not supported on %s", vmtest.TestArch())
 	}
 
-	network := qemu.NewNetwork()
-	dhcpServer, scleanup := vmtest.QEMUTest(t, &vmtest.Options{
-		Name: "TestPxeboot_Server",
-		BuildOpts: uroot.Opts{
+	var serverInitramfs string
+	var clientInitramfs string
+	if len(initramfs) == 0 {
```

Comment:
Instead of this if-statement, can you make `CreateTestInitramfs` simply return the path to the overriden initramfs if it's set? That would be a bit less logic in each test and you no longer have to pass `initramfs string` into each test.